### PR TITLE
Add eclipse-mat version 1.8.0.20180604

### PR DESCRIPTION
**Pull request by TheCjw**

### CODE REVIEW

File: eclipse-mat.json [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+        ]
+    ],
+    "checkver": {
+        "url": "https://www.eclipse.org/mat/downloads.php",
+        "re": "MemoryAnalyzer-([\\d.]+)-win32"
+    },
+    "autoupdate": {
+        "architecture": {
+            "64bit": {
+                "url": "http://download.eclipse.org/mat/1.8/rcp/MemoryAnalyzer-$version-win32.win32.x86_64.zip"
```
Comment (Ash258) [inline eclipse-mat.json]: That 1.8 inside URL is changing with Version? 
I suggest to use `$majorVersion.$minorVersion/rcp`

**Comment by TheCjw** [comment]
@Ash258 FIXED.

**Comment by TheCjw** [comment]
@se35710 Great suggestions!

**Comment by Ash258** [comment]
Could be hash universal like:
```json
"architecture":{
},
"hash": {
    "url": "https://www.eclipse.org/downloads/sums.php?file=/mat/$majorVersion.$minorVersion/rcp/$basename&type=sha512",
    "find": "([a-fA-F0-9]{128})"
}
```

**Comment by se35710** [comment]
@Ash258 There is an issue with `$basename`, it will use download.php as the filename, so the match will fail.

@TheCjw You don't need to add `#/dl.7z`, scoop will handle the extraction anyway.

**Comment by TheCjw** [comment]
@se35710 `#/dl.7z` has been removed.

**Pull request merged**